oauth2client/client.py

```diff
@@ -469,6 +467,11 @@ def clean_headers(headers):
     return clean
 
 
+def _parse_expiry(expiry):
+    if expiry and isinstance(expiry, datetime.datetime):
+        return expiry.strftime(EXPIRY_FORMAT)
```

Comment:
Never mix returning-an-expression and implicitly-return-None-by-naked-return-statement-or-falling-off-the-bottom-of-the-body in the same behavior. If you have any return statement of the form `return <expression>` anywhere in a behavior, then all of your return statements should be of the form `return <expression>` and execution should not be able to reach the end of the function without hitting a return statement.

In this case, place your `return None` inside an `else` rather than putting it at behavior top-level.